docs-go/models.md

```diff
@@ -0,0 +1,263 @@
+  fmt.Println(responseText)
+  ```
+
+You can pass options along with the model call. The options that are supported
+depend on the model and its API.
+
+* {Go}
+
+  ```go
+  request := ai.GenerateRequest{
+    Messages: []*ai.Message{
+      {Content: []*ai.Part{ai.NewTextPart("Tell me a joke about dogs.")}},
+    },
+    Config: ai.GenerationCommonConfig{
+      Temperature:     1.67,
+      StopSequences:   []string{"abc"},
+      MaxOutputTokens: 3,
+    },
+  }
+  ```
+
+### Streaming responses
+
+Genkit supports chunked streaming of model responses:
+
+* {Go}
+
+  To use chunked streaming, pass a callback function to `Generate()`:
+
+  ```go
+  request := ai.GenerateRequest{Messages: []*ai.Message{
+    {Content: []*ai.Part{ai.NewTextPart("Tell a long story about robots and ninjas.")}},
+  }}
+  response, err := ai.Generate(
+    context.Background(),
+    gemini15pro,
+    &request,
+    func(ctx context.Context, grc *ai.GenerateResponseChunk) error {
+      text, err := grc.Text()
+      if err == nil {
+        fmt.Printf("Chunk: %s\n", text)
+      }
+      return err
+    })
+
+  // You can also still get the full response.
+  responseText, err := response.Text()
+  fmt.Println(responseText)
+  ```
+
+## Multimodal input
+
+If the model supports multimodal input, you can pass image prompts:
+
+* {Go}
+
+  ```go
+  imageBytes, err := os.ReadFile("img.jpg")
+  encodedImage := base64.StdEncoding.EncodeToString(imageBytes)
+
+  request := ai.GenerateRequest{Messages: []*ai.Message{
+    {Content: []*ai.Part{
+      ai.NewTextPart("Describe the following image."),
+      ai.NewMediaPart("", "data:image/jpeg;base64,"+encodedImage),
+    }},
+  }}
+  response, err := ai.Generate(context.Background(), gemini15pro, &request, nil)
+  ```
+
+  <!-- TODO: gs:// wasn't working for me. HTTP? -->
+
+The exact format of the image prompt (`https` URL, `gs` URL, `data` URI) is
+model-dependent.
+
+## Function calling (tools)
+
+Genkit models provide an interface for function calling, for models that support
+it.
+
+* {Go}
+
+  ```go
+  myJoke := &ai.ToolDefinition{
+    Name:        "myJoke",
+    Description: "useful when you need a joke to tell",
+    InputSchema: make(map[string]any),
+    OutputSchema: map[string]any{
+      "joke": "string",
+    },
+  }
+  ai.DefineTool(
+    myJoke,
+    nil,
+    func(ctx context.Context, input map[string]any) (map[string]any, error) {
+      return map[string]any{"joke": "haha Just kidding no joke! got you"}, nil
+    },
+  )
+
+  request := ai.GenerateRequest{
+    Messages: []*ai.Message{
+      {Content: []*ai.Part{ai.NewTextPart("Tell me a joke.")},
+        Role: ai.RoleUser},
+    },
+    Tools: []*ai.ToolDefinition{myJoke},
+  }
+  response, err := ai.Generate(context.Background(), gemini15pro, &request, nil)
+  ```
+
+This will automatically call the tools in order to fulfill the user prompt.
+
+<!-- TODO: returnToolRequests: true` -->
+
+### Adding retriever context
```

Comment:
Does the Go library have this? I couldn't find the context field being used anywhere